app/src/main/java/com/amaze/filemanager/ui/fragments/preference_fragments/PrefFrag.java

```diff
@@ -231,12 +241,14 @@ public boolean onPreferenceClick(Preference preference) {
                   sharedPref
                       .edit()
                       .putInt(PreferencesConstants.PREFERENCE_DRAG_AND_DROP_PREFERENCE, which)
+                      .putString(PreferencesConstants.PREFERENCE_DRAG_AND_DROP_SUMMARY, getDragDropSummary(which))
                       .apply();
                   sharedPref
                       .edit()
                       .putString(PreferencesConstants.PREFERENCE_DRAG_AND_DROP_REMEMBERED, null)
                       .apply();
                   dialog.dismiss();
+                  restartPC(getActivity());
```

Comment:
Why do you want to restart?

Reply:
settings page doesnot reflect the selected option as summary immediately after clicking on the option you need to refresh the page same as setting a theme

Reply:
Are you can't just invalidate the specific preference?

Reply:
I dont think so, the preferences are loaded from xml file in oncreatepreferences. In order to reflect that selected option the preferences needs to be loaded again from xml file. and as its loaded on oncreatepreferences the page needs to be refreshed in order to reflect the selected option as summary